pkg/sources/generator/tickgen.go

```diff
@@ -198,6 +199,13 @@ loop:
 		// we implement Read With Wait semantics
 		select {
 		case r := <-mg.srcChan:
+			// when the channel is closed and empty, go will still read from the channel once and return the zero value.
+			// exclude the zero value from being passed to the next vertex,
+			// ensuring all messages produced by generator follow the same data schema.
+			if util.IsZeroStruct(r) {
```

Comment:
Why is it `zerostruct`?

Reply:
Thanks! I updated the PR description and addressed the comment. Don't need the isZero check, just use the second parameter should work.